improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py

```diff
@@ -30,516 +30,369 @@
-        expected = np.array(
+
+def test_cpb_coordinate_no_single_value_bins():
+    """Test the probability_bins coordinate has the expected values and
+    type with no single value lower and upper bins."""
+    expected_bounds = np.array([[0, 0.5], [0.5, 1]])
+    expected_points = np.mean(expected_bounds, axis=1)
+    plugin = Plugin(n_probability_bins=2,)
+    result = plugin._create_probability_bins_coord()
+
+    assert isinstance(result, iris.coords.DimCoord)
+    assert_allclose(result.points, expected_points)
+    assert_allclose(result.bounds, expected_bounds)
+
+
+def test_cpb_coordinate_single_value_bins():
+    """Test the probability_bins coordinate has the expected values and
+    type when using the single value lower and upper bins."""
+    expected_bounds = np.array(
+        [
+            [0.0000000e00, 1.0000000e-06],
+            [1.0000001e-06, 4.9999997e-01],
+            [5.0000000e-01, 9.9999893e-01],
+            [9.9999899e-01, 1.0000000e00],
+        ]
+    )
+    expected_points = np.mean(expected_bounds, axis=1)
+    plugin = Plugin(
+        n_probability_bins=4,
+        single_value_lower_limit=True,
+        single_value_upper_limit=True,
+    )
+    result = plugin._create_probability_bins_coord()
+
+    assert isinstance(result, DimCoord)
+    assert_allclose(result.points, expected_points)
+    assert_allclose(result.bounds, expected_bounds)
+
+
+def test_crt_coordinates():
+    """Test the reliability table coordinates have the expected values and
+    type."""
+    expected_indices = np.array([0, 1, 2], dtype=np.int32)
+    expected_names = np.array(
+        ["observation_count", "sum_of_forecast_probabilities", "forecast_count"]
+    )
+    index_coord, name_coord = Plugin()._create_reliability_table_coords()
+
+    assert isinstance(index_coord, DimCoord)
+    assert isinstance(name_coord, AuxCoord)
+    assert_array_equal(index_coord.points, expected_indices)
+    assert_array_equal(name_coord.points, expected_names)
+
+
+def test_metadata_with_complete_inputs(forecast_grid, expected_attributes):
+    """Test the metadata returned is complete and as expected when the
+    forecast cube contains the required metadata to copy."""
+    forecast_1 = forecast_grid[0]
+    forecast_1.attributes["institution"] = "Kitten Inc"
+    expected_attributes["institution"] = "Kitten Inc"
+    result = Plugin._define_metadata(forecast_1)
+    assert isinstance(result, dict)
+    assert result == expected_attributes
+
+
+def test_metadata_with_incomplete_inputs(forecast_grid, expected_attributes):
+    """Test the metadata returned is complete and as expected when the
+    forecast cube does not contain all the required metadata to copy."""
+    forecast_1 = forecast_grid[0]
+    result = Plugin._define_metadata(forecast_1)
+    assert isinstance(result, dict)
+    assert result == expected_attributes
+
+
+@pytest.mark.parametrize(
+    "input_cube, expected_shape",
```

Comment:
This works OK, but the need to use string parameters and then `getfixturevalue` inside the test function is a little messy.
As mentioned in another comment, it'd be nice if fixtures could be used directly in a parametrize, but that doesn't exist.

The main alternative I can think of would be to create a [parametrized fixture](https://docs.pytest.org/en/6.2.x/fixture.html#parametrizing-fixtures) which gathers the relevant other fixtures together.
For example:
```python

RelTableInputs = namedtuple("RelTableInputs", ["forecast", "truth", "expected_shape"])

@pytest.fixture(params=["grid", "spot"])
def create_rel_table_inputs(request, forecast_grid, forecast_spot, truth_grid, truth_spot, expected_table_shape_grid, expected_table_shape_spot):
    if request.param == "grid":
        return RelTableInputs(forecast=forecast_grid, truth=truth_grid, expected_shape=expected_table_shape_grid)
    return RelTableInputs(forecast=forecast_spot, truth=truth_spot, expected_shape=expected_table_shape_spot)
```

This is more code to set up, but it would avoid duplicated lines between this test function and others below that want to test with both grid and site.
I'm not sure what's better overall - interested in others opinions about this.

Reply:
Thanks, I think this will may be cleaner and make less code overall.

Reply:
I noticed that fixture parametrization made the tests a little bit slower so timed them out of curiosity:

Parametrization:
```
0.04s setup    improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[grid]
0.03s setup    improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[spot]
```
Fixture parametrization:
```
0.04s setup    improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[grid]
0.04s setup    improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[spot]
0.02s call     improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[spot]
0.02s call     improver_tests/calibration/reliability_calibration/test_ConstructReliabilityCalibrationTables.py::test_prb_table_values[grid]
```
It doesn't make enough of a difference to matter but just thought I would share.